src/createReducer.ts

```diff
@@ -0,0 +1,56 @@
+import createNextState from 'immer'
+import { AnyAction, Action, Reducer } from 'redux'
+
+/**
+ * An *case reducer* is a reducer function for a speficic action type. Case
+ * reducers can be composed to full reducers using `createReducer()`.
+ *
+ * Unlike a normal Redux reducer, a case reducer is never called with an
+ * `undefined` state to determine the initial state. Instead, the initial
+ * state is explicitly specified as an argument to `createReducer()`.
+ *
+ * In addition, a case reducer can choose to mutate the passed-in `state`
+ * value directly instead of returning a new state. This does not actually
+ * cause the store state to be mutated directly; instead, thanks to
+ * [immer](https://github.com/mweststrate/immer), the mutations are
+ * translated to copy operations that result in a new state.
+ */
+export type CaseReducer<S = any, A extends Action = AnyAction> = (
```

Comment:
Are redux _states_ required to be objects, or can they be primitive types too? If they have to be objects, `S extends object`.

Reply:
Redux states can be anything, not just strings.